File: javascript/datatypes.js
function displayDataTypes() {
  console.log("Size of Data Types in JavaScript");
  console.log("===============================");
  console.log(`Size of Number = ${8} Bytes`); // JavaScript number is always a 64-bit double precision float
  console.log(
    `Size of String = Depends on the length of the string (approximated as 2 bytes per character)`
  );
  console.log(`Size of Boolean = ${typeof true} (Internally stored as 1 byte)`);
  console.log(`Size of Object = Depends on the number of properties`);
  console.log(
    `Size of Undefined = Typically 0 Bytes (though variable size in memory allocation)`
  );
  console.log("\n");

  console.log("The Range of Data Types in JavaScript");
  console.log("===============================");
  console.log("Range of Number: From -1.8e308 to 1.8e308 (approximately)");
  console.log("Range of Boolean: Only two values: true and false");
  console.log("Range of String: No fixed range, depends on memory available.");
  console.log(
    "Range of Object: No fixed range, depends on properties and references."
  );
  console.log(
    "Range of Undefined: Undefined has no range, it's a special type."
  );
}

displayDataTypes();

File: python/datatypes.py
import sys

def main():
    # Displaying sizes of data types
    print("Size of Data Types in Python")
    print("===========================")
    print(f"Size of int = {sys.getsizeof(0)} Bytes")
    print(f"Size of float = {sys.getsizeof(0.0)} Bytes")
    print(f"Size of complex = {sys.getsizeof(complex(0, 0))} Bytes")
    print(f"Size of str = {sys.getsizeof('')} Bytes")
    print(f"Size of bool = {sys.getsizeof(True)} Bytes")
    print(f"Size of list = {sys.getsizeof([])} Bytes")
    print("\n")

    # Displaying ranges of numeric types
    print("The Range of Numeric Types in Python")
    print("=============================")
    print("Range of int: No fixed range, can grow as large as memory allows.")
    print("Range of float: From -1.8e308 to 1.8e308 (approximately)")
    print("Range of complex: Complex numbers are unbounded")
    print("Range of bool: Only two values: True and False")
    print("\n")

if __name__ == "__main__":
    main()